js_modules/dagit/src/configeditor/codemirror-yaml/mode.tsx

```diff
@@ -400,26 +406,30 @@ function findAutocomplete(
     for (const parent of parents) {
       const parentTypeDef = available.find(({ name }) => parent.key === name);
 
-      if (!parentTypeDef || !parentTypeDef.configType.name) {
+      if (!parentTypeDef) {
         return [];
       }
 
-      let childTypeName = parentTypeDef.configType.name;
-      let childEntiresUnique = true;
+      let childTypeKey = parentTypeDef.configType.key;
+      let childEntriesUnique = true;
 
-      if (parentTypeDef.configType.name.startsWith("List.")) {
-        childTypeName = parentTypeDef.configType.name.substr(5);
-        childEntiresUnique = false;
+      if (parentTypeDef.configType.isList) {
+        // ofType guaranteed to have value in the List case
+        // better way to enforce this?
+        if (parentTypeDef.configType.ofType) {
```

Comment:
Ahh this is interesting. Technically you could consolidate these two lines into `if (parentTypeDef.configType.__typename == "ListConfigType")` because TS knows that `ofType` is included in the `ListConfigType` case, but there's no known relationship between `ofType` and `isList`... I think that makes a strong case for the __typename solution rather than `isList`, etc., though that's not a great look either...

The only other solution I can think of would be to make `isList` something like `isListOfType`, where the same field is either null (indicating it's not a list) or non-null with a type payload.

Reply:
So this is actually the handcrafted type `TypeConfig` rather than the apollo autogenerated stuff.

```js
export type TypeConfig = {
  rootTypeKey: string;
  types: {
    [key: string]: {
      fields: Array<{
        name: string;
        configType: {
          __typename: string;
          isList: boolean;
          isNullable: boolean;
          key: string;
          name: string | null;
          ofType?: {
            key: string;
          };
        };
      }>;
    };
  };
};
```

These don't have the discriminated unions that that generated apollo types have so this doesn't work out of the box. I spent a few minutes trying to mimic this in the TypeConfig code but decided that was worth more trouble than it was worth for now.